allennlp/nn/loss/shared_norm_loss.py

```diff
@@ -0,0 +1,46 @@
+import torch
+from torch.autograd import Variable
+
+from allennlp.nn.util import logsumexp
+
+# span_logits is (batch_size, num_paragraphs, num_tokens)
+# paragraph_mask is (batch_size, num_paragraphs)
+# actual_spans is (batch_size, num_paragraphs, num_spans)
```

Comment:
Its a small bit confusing that this is called span, 
Even though it is not a span and is in fact only the start or end. 
But when there is documentation on the class it will probably be clear.